realm/realm-library/src/objectServer/java/io/realm/ObjectServer.java

```diff
@@ -87,12 +81,12 @@ public static void init(Context context, String appDefinedUserAgent) {
                             "Directory '%s' for SyncManager cannot be created. ",
                             dir.getPath()));
                 }
-                SyncManager.nativeInitializeSyncManager(dir.getPath(), userAgent);
+                SyncManager.nativeInitializeSyncManager(dir.getPath(), userAgentBindingInfo, appDefinedUserAgent);
```

Comment:
where is `appUserAgentInfo` populated?

Reply:
It is an optional setting set through `Realm.init(context, userAgent)`

Reply:
is this exposed for test? who needed this override?

Reply:
It was added in this PR: https://github.com/realm/realm-java/pull/6270
It looks like we don't have automatic tests for it, but only verified the behavior manually.